### PR TITLE
Small bugs, string in k-means, add raw_data var.

### PR DESCRIPTION
**Pull request by gregori0o**
I fixed small bugs in visualization of results section.

I changed distance function in k-means algorithm. Now, k-means algorithm skip column with no numeric data. It's good for presentation, because we have column with strings in result section. Previous implementation works only with categories data. I think we will add some functionality in preprocessing, which changes string into numeric.

### CODE REVIEW

File: src/widgets/unfold_widgets/preprocessing_widget.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def _clear_plot(self):
 
     def add_columns_to_layout(self):
         self.clear_column_layout()
-        columns = self.engine.get_columns()
+        columns = self.engine.get_raw_columns()
```
Comment (t0maboro) [inline src/widgets/unfold_widgets/preprocessing_widget.py:157]: I really enjoy that solution, but here I am not a fan of the previous solution with clearing checkboxes after submitting. Maybe we can keep a checkbox state after clicking select actually? Talking about line 160.

**Review by t0maboro — APPROVED** [review]
I appreciate this idea. Just one thing to consider, nothing critical. @mhawryluk opinion about the suggestion below would be nice - because it is maybe only my feeling. Anyway, can be merged.

File: src/algorithms/clustering/k_means.py [hunk 2]
```diff
@@ -81,15 +82,16 @@ def mean(self, group: pd.DataFrame) -> Tuple:
...
-                counter = {}
-                most_frequent = None
-                count = 0
-                for element in column:
-                    counter[element] = counter.get(element, 0) + 1
-                    if counter[element] > count:
-                        count = counter[element]
-                        most_frequent = element
-                result.append(most_frequent)
+                result.append("NULL")
```
Comment (mhawryluk) [inline src/algorithms/clustering/k_means.py]: will changing "NULL" string to _None_ break something down the line? if not then that would probably be better, but I assume that this part is temporary either way, so it doesn't matter much

File: src/widgets/results_widgets/k_means_results.py [hunk 3]
```diff
@@ -175,8 +191,16 @@ def update_cluster_plot(self):
             sep_x = 0.1 * (max_x - min_x)
             sep_y = 0.1 * (max_y - min_y)
 
-            x_centroids = self.centroids[self.ox]
-            y_centroids = self.centroids[self.oy]
+            # x_centroids = self.centroids[self.ox]
+            # y_centroids = self.centroids[self.oy]
```
Comment (mhawryluk) [inline src/widgets/results_widgets/k_means_results.py]:
```suggestion
```

File: src/widgets/results_widgets/k_means_results.py [hunk 4]
```diff
@@ -158,10 +166,18 @@ def update_cluster_plot(self):
             sep_x = 0.1 * (max_x - min_x)
             sep_y = 0.1 * (max_y - min_y)
 
-            x_centroids = self.centroids[self.ox]
-            y_centroids = self.centroids[self.oy]
-
-            self.centroids_canvas.chosen_centroid_plot(x, y, None, None, x_centroids.iloc[self.selected_cluster],
+            # x_centroids = self.centroids[self.ox]
+            # y_centroids = self.centroids[self.oy]
```
Comment (mhawryluk) [inline src/widgets/results_widgets/k_means_results.py]:
```suggestion
```

File: src/widgets/unfold_widgets/preprocessing_widget.py (hunk 1 above)
Comment (mhawryluk) [reply src/widgets/unfold_widgets/preprocessing_widget.py:157]: I agree, they shouldn't be cleared after select. Ideally there should be no button, but rather selected columns are gathered when switching tabs. Also there could be a select-all/unselect-all button. But that's for another day.

**Review by mhawryluk — APPROVED** [review]
Great job!

**Pull request merged**